### PR TITLE
chore: improve logging by adding shadow versions

### PR DESCRIPTION
**Pull request by nikkhilmuthye**
**Issue #, if available:**

**Description of changes:**
Improve logging to add in the cloud and local shadow versions.

**Why is this change necessary:**
Helps the customer to understand which version of shadow is being updated or synced.

**How was this change tested:**

**Any additional information or context required to review the change:**

**Checklist:**
 - [ ] Updated the README if applicable
 - [ ] Updated or added new unit tests
 - [ ] Updated or added new integration tests
 - [ ] If your code makes a remote network call, it was tested with a proxy
 
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/aws/greengrass/shadowmanager/sync/model/LocalDeleteSyncRequest.java [hunk 1]
```diff
@@ -89,18 +91,40 @@ public void execute(SyncContext context) throws SkipSyncRequestException, Unknow
...
+                        .kv(LOG_CLOUD_VERSION_KEY, deletedCloudVersion + 1)
+                        .log("Successfully deleted local shadow");
 
             } catch (ResourceNotFoundError e) {
                 logger.atInfo()
                         .kv(LOG_THING_NAME_KEY, getThingName())
                         .kv(LOG_SHADOW_NAME_KEY, getShadowName())
+                        .kv(LOG_LOCAL_VERSION_KEY, syncInformation.getLocalVersion())
+                        .kv(LOG_CLOUD_VERSION_KEY, syncInformation.getCloudVersion())
+                        .kv("deleted-cloud-version", deletedCloudVersion)
```
Comment (youtuyy) [inline src/main/java/com/aws/greengrass/shadowmanager/sync/model/LocalDeleteSyncRequest.java:107]: Make the key a new constant?

File: src/main/java/com/aws/greengrass/shadowmanager/sync/model/LocalUpdateSyncRequest.java [hunk 2]
```diff
@@ -132,13 +134,32 @@ public void execute(SyncContext context) throws SkipSyncRequestException, Confli
                         .lastSyncTime(updateTime)
                         .cloudDeleted(false)
                         .build());
+                logger.atDebug()
+                        .kv(LOG_THING_NAME_KEY, getThingName())
+                        .kv(LOG_SHADOW_NAME_KEY, getShadowName())
+                        .kv(LOG_LOCAL_VERSION_KEY, localUpdatedVersion)
```
Comment (youtuyy) [inline src/main/java/com/aws/greengrass/shadowmanager/sync/model/LocalUpdateSyncRequest.java:140]: localUpdatedVersion will be a boolean?

**Review by youtuyy — APPROVED** [review]

File: src/main/java/com/aws/greengrass/shadowmanager/sync/model/LocalUpdateSyncRequest.java [hunk 3]
```diff
@@ -121,24 +124,44 @@ public void execute(SyncContext context) throws SkipSyncRequestException, Confli
 
                 byte[] updatedDocument = response.getCurrentDocument();
                 long updateTime = Instant.now().getEpochSecond();
+                long localUpdatedVersion = getUpdatedVersion(response.getUpdateThingShadowResponse().getPayload())
+                        .orElse(currentLocalVersion + 1);
```
Comment (randolphtan121) [inline src/main/java/com/aws/greengrass/shadowmanager/sync/model/LocalUpdateSyncRequest.java:128]: What are the situations in which the `orElse` gets used?

Comment (nikkhilmuthye) [reply src/main/java/com/aws/greengrass/shadowmanager/sync/model/LocalUpdateSyncRequest.java:128]: It is just a sanity check. Should be fine to have it.

**Review by randolphtan121 — APPROVED** [review]

**Pull request merged**